main.tf

```diff
@@ -0,0 +1,47 @@
+provider "aws" {
+  region = var.credentials.region
+  access_key = var.credentials.access_key
+  secret_key = var.credentials.secret_key
+}
+
+resource "aws_instance" "demo_ec2" {
+  ami           = var.ec2.ami
+  instance_type = var.ec2.instance_type
+  security_groups =  [aws_security_group.bastion_host.name]
+  key_name   = aws_key_pair.demokeypair.key_name
+  tags = {
+    Name = "${var.prefix}-bastion-host"
+  }
+}
+
+resource "aws_security_group" "bastion_host" {
+  name = "${var.prefix}-bastion-host" 
+  description = "security group access for all" 
+  
+ingress {
+    cidr_blocks = var.ingress.cidr_blocks
+    from_port = var.ingress.from_port
+    to_port = var.ingress.from_port
+    protocol = var.ingress.protocol
+  }
+egress {
+    cidr_blocks = var.egress.cidr_blocks
+    from_port = var.egress.from_port
+    to_port = var.egress.from_port
+    protocol = var.egress.protocol
+ }
+}
+
+resource "aws_key_pair" "demokeypair" {
+  key_name   = "${var.prefix}-KeyPair"

```

Comment:
naming convention for aws:
1. lowercase letters
2. use dash "-" to separate someDayWillBeSunny, some-day-will-be-sunny

Reply:
Fixed naming